L1Trigger/L1THGCal/plugins/concentrator/HGCalConcentratorProcessorSelection.cc

```diff
@@ -9,26 +9,36 @@ HGCalConcentratorProcessorSelection::HGCalConcentratorProcessorSelection(const e
     : HGCalConcentratorProcessorBase(conf),
       fixedDataSizePerHGCROC_(conf.getParameter<bool>("fixedDataSizePerHGCROC")),
       coarsenTriggerCells_(conf.getParameter<bool>("coarsenTriggerCells")) {
-  std::string selectionType(conf.getParameter<std::string>("Method"));
-  if (selectionType == "thresholdSelect") {
-    selectionType_ = thresholdSelect;
-    thresholdImpl_ = std::make_unique<HGCalConcentratorThresholdImpl>(conf);
-  } else if (selectionType == "bestChoiceSelect") {
-    selectionType_ = bestChoiceSelect;
-    bestChoiceImpl_ = std::make_unique<HGCalConcentratorBestChoiceImpl>(conf);
-  } else if (selectionType == "superTriggerCellSelect") {
-    selectionType_ = superTriggerCellSelect;
-    superTriggerCellImpl_ = std::make_unique<HGCalConcentratorSuperTriggerCellImpl>(conf);
-  } else if (selectionType == "mixedBestChoiceSuperTriggerCell") {
-    selectionType_ = mixedBestChoiceSuperTriggerCell;
-    bestChoiceImpl_ = std::make_unique<HGCalConcentratorBestChoiceImpl>(conf);
-    superTriggerCellImpl_ = std::make_unique<HGCalConcentratorSuperTriggerCellImpl>(conf);
-  } else {
+  std::vector<std::string> selectionType(conf.getParameter<std::vector<std::string>>("Method"));
+  if (selectionType.size() != nSubDetectors_) {
     throw cms::Exception("HGCTriggerParameterError")
-        << "Unknown type of concentrator selection '" << selectionType << "'";
+      << "Inconsistent number of sub-detectors (should be " << nSubDetectors_ << ")";
   }
-
-  if (coarsenTriggerCells_ || fixedDataSizePerHGCROC_) {
+  selectionType_.resize(nSubDetectors_);
```

Comment:
Since `nSubdetectors_` is an hardcoded constant, `selectionType_` could be directly constructed with the correct size